src/vs/platform/windows/electron-main/windowsStateHandler.ts

```diff
@@ -227,6 +229,8 @@ export class WindowsStateHandler extends Disposable {
 			return; // during quit, many windows close in parallel so let it be handled in the before-quit handler
 		}
 
+		this.loggerMainService.deregisterLoggerResources(window.id);
```

Comment:
This is not a good location to do cleanup, the intent of this class is to persist window state. Why not have this in the windowImpl?

Reply:
Sure, moved it. Thanks.